ui/common/src/socket.ts

```diff
@@ -41,47 +41,50 @@ interface Settings {
   params?: Partial<Params>;
   options?: Partial<Options>;
 }
-
-const origSend = WebSocket.prototype.send;
+// TODO - find out why are there three different types of settings
 
 const isOnline = () => !('onLine' in navigator) || navigator.onLine;
 
-// versioned events, acks, retries, resync
-export default class StrongSocket {
-  settings: Settings;
-  options: Options;
-  version: number | false;
-  ws: WebSocket | undefined;
-  pingSchedule: Timeout;
-  connectSchedule: Timeout;
-  ackable: Ackable = new Ackable((t, d, o) => this.send(t, d, o));
-  lastPingTime: number = performance.now();
-  pongCount = 0;
+export default class StrongSocket implements SocketI {
   averageLag = 0;
-  tryOtherUrl = false;
-  autoReconnect = true;
-  nbConnects = 0;
-  storage: LichessStorage = storage.make(
-    document.body.dataset.socketAlternates ? 'surl-alt' : 'surl17',
-    30 * 60 * 1000,
-  );
+
+  private settings: Settings;
+  private options: Options;
+  private version: number | false;
+  private ws: WebSocket | undefined;
+  private pingSchedule: Timeout;
+  private connectSchedule: Timeout;
+  private ackable: Ackable = new Ackable((t, d, o) => this.send(t, d, o));
+  private lastPingTime: number = performance.now();
+  private pongCount = 0;
+  private tryOtherUrl = false;
+  private storage: LichessStorage = storage.make('surl17', 30 * 60 * 1000);
   private _sign?: string;
   private resendWhenOpen: [string, any, any][] = [];
-  private baseUrls = (document.body.dataset.socketAlts || document.body.dataset.socketDomains!).split(',');
-  private static defaultOptions: Options = {
-    idle: false,
-    pingMaxLag: 9000, // time to wait for pong before resetting the connection
-    pingDelay: 2500, // time between pong and ping
-    autoReconnectDelay: 3500,
-    protocol: location.protocol === 'https:' ? 'wss:' : 'ws:',
-    isAuth: document.body.hasAttribute('data-user'),
-  };
+  private baseUrls = document.body.dataset.socketDomains!.split(',');
+
+  private lastUrl?: string;
+  private heartbeat = browserTaskQueueMonitor(1000);
+  private isTestUser = document.body.dataset.socketTestUser === 'true';
+  private isTestRunning = document.body.dataset.socketTestRunning === 'true';
+  private stats: { store?: ObjectStorage<any>, m2: number, n: number, mean: number } =
+    { m2: 0, n: 0, mean: 0 };
 
   constructor(
     readonly url: string,
     version: number | false,
     settings: Partial<Settings> = {},
   ) {
+    this.options = {
```

Comment:
i moved initialization of the various options, settings, params to the constructor as they are all instance related, whether or not they're hard coded or set by constructor arguments